### PR TITLE
Add pacemaker(pacemaker-based) resource translatefile

### PR DESCRIPTION
**Pull request by daaitudian**
In order to realize the support of Pacemaker for Chinese, the internationalization method was added, and the internationalization modification was gradually carried out based on Pacemaker -based.

### CODE REVIEW

**Comment by daaitudian** [comment]
ok to test，we have a team doing research and development on this, so we want to be actively involved in community building, and internationalization is one of our attempts.

**Comment by wenningerk** [comment]
test this please

**Comment by daaitudian** [comment]
I would like to know what is the problem, can we test it locally?

**Comment by wenningerk** [comment]
> I would like to know what is the problem, can we test it locally?

You should be able to click through the ci-results till:

11:09:32 msgfmt po/pacemaker.po -o /usr/share/locale/zh_CN/LC_MESSAGES/pacemaker.mo
11:09:32 msgfmt: error while opening "po/pacemaker.po" for reading: No such file or directory
11:09:32 make[1]: *** [Makefile:1103: install] Error 1
11:09:32 make[1]: Leaving directory '/srv/workspace/pacemaker-build-all-voting/pacemaker-build-all-voting/centos-8-x86-64/pacemaker-4c60110a4/_build/sub'

Guess that is an out-of-tree-build issue and the files need to go into EXTRA_DIST.

File: po/pacemaker_en.po [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+"Project-Id-Version: PACKAGE VERSION\n"
+"Report-Msgid-Bugs-To: \n"
+"POT-Creation-Date: 2020-12-11 15:59+0800\n"
+"PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE\n"
+"Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
+"Language-Team: LANGUAGE <LL@li.org>\n"
+"Language: \n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Transfer-Encoding: 8bit\n"
```
Comment (kgaillot) [inline po/pacemaker_en.po:18]: If en is the default, do we really need a po file for it?

**Review by kgaillot — COMMENTED** [review]
Hi,

Thanks for this effort! This would be a major enhancement to Pacemaker. I am sure a lot of users would appreciate it.

However, I have some concerns.

Commercially supported distributions have a difficult trade-off between supporting locales in log messages, to make it easier for users themselves, or not supporting locales, to make it easier for their support personnel to investigate support cases. Using the configure options as described in my review will allow distributions and users who build from source to choose whether to enable locale support. If major distributions choose to disable the support, that will reduce the impact of adding the feature.

The biggest challenge of localization is finding translators committed to putting in the work to update translations for every release. Pacemaker makes a lot of message changes in each release. We can add the support and see how much response we get, but over the long term, if translators don't keep it up to date, we'll have to drop it. Also, we will need to find independent volunteers to serve as proofreaders for each translation.

The barrier of having to learn git and command-line tools discourages many potential translators. We could consider using a service like Zanata to make it easier. This would not have to be done now; we could add the ability later if people are interested. This is an interesting description of how one large project integrated with Zanata:

https://www.berrange.com/posts/2018/11/29/improved-translation-po-file-handling-by-ditching-gettext-autotools-integration/

**Comment by daaitudian** [comment]
Thank you very much for your help. We attach great importance to your questions and will consider them carefully in the future. It is very likely to be the direction of our team's future research and development, especially Zanata you mentioned. But at this stage, we need a little time to upload the current version.

**Comment by daaitudian** [comment]
Thank you very much for your help. We attach great importance to your questions and will consider them carefully in the future. It is very likely to be the direction of our team's future research and development, especially Zanata you mentioned. But at this stage, we need a little time to upload the current version.

**Pull request closed without merging**